exercises/nucleotide-count/canonical-data.json

```diff
@@ -39,12 +62,38 @@
           }
         },
         {
-          "description": "strand with invalid nucleotides",
+          "description": "counts all nucleotides correctly in mixed case.",
           "property": "nucleotideCounts",
-          "strand": "AGXXACT",
+          "strand": "aGcTtTtCaTtCtGaCtGcAaCgGgCaAtAtGtCtCtGtGtGgAtTaAaAaAaGaGtGtCtGaTaGcAgC",
```

Comment:
This exercise is not about case comparisons. We should be able to trust that all the input characters are valid.

Reply:
I respectfully disagree. See my reasoning above (https://github.com/exercism/problem-specifications/pull/895#discussion_r139141823).

We educate people about programming and teaching them what it means to test for all equivalence classes should be an integral part of this progress right from the beginning.

Why should we assume that all inputs are valid? This works to a certain degree in one-man projects and quick (and/or dirty) implementations, but this is nothing we should promote. In reality murphy's law rules.

I know that several test cases seem tedious to write, but why bother here? The users don't even write the tests, we do. But we should be a good example.

Reply:
In many cases when developing, it is perfectly safe to assume that an input is valid. You may for example have integration test coverage proving that invalid input never reaches your code under unit test. 

Validations may be redundant, and cost both programmer time and CPU cycles.

Reply:
I agree, we should avoid redundant tests.
However, we don't have such (integration-) tests in place.

Reply:
There are also no tests that the keyboard is plugged in or the screen is turned on.

Reply:
Now you are dragging the matter into ridiculousness. Sorry, but this is not an argument. ;-)

Reply:
> Now you are dragging the matter into ridiculousness.

Guilty. :blush:

> However, we don't have such (integration-) tests in place.

Is this not guilty of the same thing?
That we don't have integration tests is irrelevant because we do not need them, we can trust that the test input is correct and valid.

Reply:
To a degree, yes.
But I think we should continue this thread over at the discussion thread.